sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -99,47 +131,121 @@ export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData
+export interface DeviceConnectionStateEventProperties {
+    deviceConnectionStateEventInfo?: DeviceConnectionStateEventInfo;
+    deviceId?: string;
+    hubName?: string;
+    moduleId?: string;
+}
+
+// @public
+export interface DeviceLifeCycleEventProperties {
+    deviceId?: string;
+    hubName?: string;
+    twin?: DeviceTwinInfo;
+}
+
+// @public
+export interface DeviceTelemetryEventProperties {
+    body?: any;
+    properties?: {
+        [propertyName: string]: string;
+    };
+    systemProperties?: {
+        [propertyName: string]: string;
+    };
 }
 
 // @public
-export type EventGridClientOptions = PipelineOptions;
+export interface DeviceTwinInfo {
+    authenticationType?: string;
+    cloudToDeviceMessageCount?: number;
+    connectionState?: string;
+    deviceId?: string;
+    etag?: string;
+    lastActivityTime?: string;
+    properties?: DeviceTwinInfoProperties;
+    status?: string;
+    statusUpdateTime?: string;
+    version?: number;
+    x509Thumbprint?: DeviceTwinInfoX509Thumbprint;
+}
+
+// @public
+export interface DeviceTwinInfoProperties {
+    desired?: DeviceTwinProperties;
+    reported?: DeviceTwinProperties;
+}
+
+// @public
+export interface DeviceTwinInfoX509Thumbprint {
+    primaryThumbprint?: string;
+    secondaryThumbprint?: string;
+}
+
+// @public
+export interface DeviceTwinMetadata {
+    lastUpdated?: string;
+}
+
+// @public
+export interface DeviceTwinProperties {
+    metadata?: DeviceTwinMetadata;
+    version?: number;
+}
 
 // @public
 export class EventGridConsumer {
-    constructor(decoders?: Record<string, CustomEventDataDecoder>);
-    // Warning: (ae-forgotten-export) The symbol "CustomEventDataDecoder" needs to be exported by the entry point index.d.ts
-    //
+    constructor(options?: EventGridConsumerOptions);
     // (undocumented)
-    readonly customDecoders: Record<string, CustomEventDataDecoder>;
-    decodeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
-    decodeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
-    decodeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
-    decodeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+    readonly customDeserializers: Record<string, CustomEventDataDeserializer>;
+    deserializeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
+    deserializeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
+    deserializeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
+    deserializeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+}
+
+// @public
+export interface EventGridConsumerOptions {
+    customDeserializers: Record<string, CustomEventDataDeserializer>;
 }
 
 // @public
 export interface EventGridEvent<T> {
     data: T;
     dataVersion: string;
-    eventTime: Date;
+    eventTime?: Date;
     eventType: string;
-    id: string;
+    id?: string;
```

Comment:
is it better to explicitly state `id: string | undefined`?

Reply:
Done.

Reply:
Actually, not sure about this one.  After making the change some tests I had where I didn't explicitly pass an id were breaking because `id` was not defined on the object?

The goal here is that someone can pass an object literal without a specific id and in that case we generate one for them.  I think forcing the user to pass `id: undefined` to opt into this is very confusing.  Thoughts?

Reply:
It would be more work... but could we have two versions (or use some TS magic) of this interface, since I presume that that EventGridConsumer will give you an EventGridEvent that **always** has an id, but PublisherClient has optional ids? 

I recognize that our requirements on what a dev passes to us are different than what we pass to them, I just find that having to write `event.id!` or `if(event.id)` in consuming code to be annoying if I know it will always have an id prop.

Reply:
> I recognize that our requirements on what a dev passes to us are different than what we pass to them, I just find that having to write event.id! or if(event.id) in consuming code to be annoying if I know it will always have an id prop.

Ahh!  Thanks for making this explicit.  I was not thinking about it from the consumption angle.  Yeah, we should be able to do something like this.